### ISSUE TITLE
Outdated "Contribute a Java Extension" Wiki

### ISSUE THREAD

**Issue by astrelsky** [redhat-developer/vscode-java#1229]
The "Contribute a Java Extension" Wiki is outdated and impossible to follow

##### Environment
- Operating System:
- JDK version: 13.0.1
- Visual Studio Code version: 1.42.0-insider
- Java extension version: 0.55.2

##### Steps To Reproduce - Please attempt to reproduce!
1. Forget all prior knowledge of how to create a vscode-java extension.
2. Create a fresh Eclipse IDE for Java Developers installation.
3. In a clean, new project directory, attempt to follow the wiki and put together a server bundle using only the steps and information provided by the wiki and its external links.
4. Fail.
5. Repeat continuously for days/weeks.

##### Current Result
It is impossible to "Contribute a Java Extension"

##### Expected Result
A simple, easy to follow guide on how to get started creating a Java extension. A working example project would be preferred.

##### Additional Information
While this may have sounded harsh it is a result of my frustration. I find it absurd that a requirement of creating an extension to a plugin for your favorite lightweight IDE requires the installation of a resource heavy IDE which the user may loathe using.

@testforstephen if it possible to do, would you mind creating a sample project which leverages your [vscode-pde](https://github.com/testforstephen/vscode-pde) to create an extension which targets the eclipse.jdt.ls currently in use by vscode-java?

**Comment by testforstephen** [comment]
> if it is possible to do, would you mind creating a sample project which leverages your vscode-pde to create an extension targeting the eclipse.jdt.ls currently in use by vscode-java?

Fair enough. A copyable template is indeed valuable, i will take a look.

Currently if you're working on contributing a java extension based on vscode-java, you may take a look at the similar extensions: [vscode-java-dependency](https://github.com/microsoft/vscode-java-dependency), [vscode-maven](https://github.com/microsoft/vscode-maven), [vscode-java-test](https://github.com/microsoft/vscode-java-test). These extensions all are based on vscode-java extenion.

**Comment by angelozerr** [comment]
It should be fine if we could have this list in the readme
